low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -0,0 +1,411 @@
+                                "dbus loop with error: %s" % e)
+
+    def connect_to_prop_changed_signal(self, service):
+        """Connect the service to the 'PropertiesChanged' Signal."""
+        try:
+            unit = self._bus.get_object('org.freedesktop.systemd1',\
+                                        self._manager.LoadUnit(service))
+            Iunit = Interface(unit, dbus_interface='org.freedesktop.DBus.Properties')
+            state = str(Iunit.Get('org.freedesktop.systemd1.Unit', 'ActiveState'))
+            substate = str(Iunit.Get('org.freedesktop.systemd1.Unit', 'SubState'))
+            pid = str(Iunit.Get('org.freedesktop.systemd1.Service', 'ExecMainPID'))
+
+            self.update_status(service, state, substate, pid)
+
+            Iunit2 = Interface(unit,
+                            dbus_interface='org.freedesktop.systemd1.Manager')
+
+            Iunit2.connect_to_signal('PropertiesChanged',
+                                    lambda a, b, c, p = unit :
+                                    self.on_prop_changed(a, b, c, p),
+                                    dbus_interface=PROPERTIES_IFACE)
+
+            logger.debug(f"{service}({pid}) state is {state}:{substate}")
+
+            if state != "active":
+                self.failed_services.append(service)
+                self.raise_alert(service, "N/A", state, "N/A", substate,
+                                    "N/A", pid, 0)
+                logger.error(f"{service} is not active initially. state = {state}:{substate}")
+
+            self.services_to_monitor.remove(service)
+
+            return None
+        except DBusException as err:
+            return err
+
+    def check_notactive_services(self):
+        """Raise Alert if any of the not-active services has exceeded
+            the threshould time for inactivity.
+        """
+        temp_not_active_ser = self.not_active_services.copy()
+        for service, start_time in temp_not_active_ser.items():
+            logger.debug(f"{service} : {start_time}, {int(time.time()) - start_time}, {self.max_wait_time}")
+            if int(time.time()) - start_time > self.max_wait_time:
+                state = self.service_status[service]["state"]
+                substate = self.service_status[service]["substate"]
+                pid = self.service_status[service]["pid"]
+                self.not_active_services.pop(service)
+                self.failed_services.append(service)
+                self.raise_alert(service, state, state, substate, substate,
+                                pid , pid, 2)
+
+    def update_status(self, service, state, substate, pid):
+        self.service_status[service] = {
+            "state" : state,
+            "substate" : substate,
+            "pid" : pid
+        }
+
+    def on_prop_changed(self, interface, changed_properties,
+                                invalidated_properties, unit):
+        """Process the `PropertiesChanged` Signal from some service."""
```

Comment:
Handler to process the service state change signal

Reply:
done..